src/storage/src/protocol/client.rs

```diff
@@ -324,14 +328,15 @@ impl Arbitrary for StorageResponse<mz_repr::Timestamp> {
 pub struct PartitionedStorageState<T> {
     /// Number of partitions the state machine represents.
     parts: usize,
-    /// Upper frontiers for sources.
-    uppers: HashMap<GlobalId, MutableAntichain<T>>,
+    /// Upper frontiers for sources and sinks, both unioned across all partitions and from each
+    /// individual partition.
+    uppers: HashMap<GlobalId, (MutableAntichain<T>, Vec<Antichain<T>>)>,
```

Comment:
Hm, good question.

The first answer is I don't know! I simply adjusted the code that was already there.

The second answer is it looks like we need the `Partitioned` client, because storaged [can be configured to have more than one worker threads](https://github.com/MaterializeInc/materialize/blob/1b3d825023c63efa551e231743bad3cdd59e7dd3/src/storage/src/protocol/server.rs#L37-L38) and each worker thread maps to a partition.